### PR TITLE
fixes BS-181

### PR DESCRIPTION
**Pull request by jaimecasero**
**What this PR does / why we need it**:
added servlet to allow cache directory separation from jboss installation
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # https://telestax.atlassian.net/browse/BS-181

**Special notes for your reviewer**:

### CODE REVIEW

File: restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/CacheServletTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+/**
+ * Test the servlet serving the media cache content is working.
+ *
+ * The restcomm configuration will be filtered by maven to replace the cache path
+ * to be in target/test-classes
+ *
+ */
+@RunWith(Arquillian.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
```
Comment (maria-farooq) [inline restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/CacheServletTest.java]: do we really need this?

Comment (jaimecasero) [reply restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/CacheServletTest.java]: not really becuase there is one test case/method in this test class now, but if someone else adds a new one i prefer this to be the default.

File: restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/CacheServletTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    private Deployer deployer;
+    @ArquillianResource
+    URL deploymentUrl;
+
+
+    @Test
+    public void testMediaDownloading() {
+
+        Client jersey = Client.create();
+        WebResource resource = jersey.resource(getResourceUrl("/cache/sample.wav"));
```
Comment (maria-farooq) [inline restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/CacheServletTest.java]:
```
RQNT 147483687 mobicents/ivr/4@10.0.0.28:2427 MGCP 1.0
N:restcomm@10.0.0.28:2727
R:AU/oc(N),AU/of(N)
X:8
S:AU/pa(an=https://staging.restcomm.com:443/restcomm/cache/ACae6e420f425248d6a26948c17a9e2acf/101cc31d1b07e8285c40d13d83f63da9caaaf21155a321f231980601bac555de.wav it=1)
```
@jaimecasero i am check current command sent by RC to RMS for playing a file
i notice `/ACae6e420f425248d6a26948c17a9e2acf/`
i dont think we should have account sids in url but sharing this to see the impact (if any)

Comment (maria-farooq) [reply restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/CacheServletTest.java]: https://github.com/RestComm/Restcomm-Connect/blob/master/restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java#L565

Comment (jaimecasero) [reply restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/CacheServletTest.java]: well how we compose URL is not really a concern of this servlet. This servlet only concern is serve/expose  the content outside RC, so RMS may consume it. 

Having said that,we can review the URL composition model in the cache. WE surely need to phisically distribute the files in the dir, we dont want all files to be in same dir, we could hit a OS limit of files in dir soon.

**Comment by jaimecasero** [comment]
@maria-farooq  fair challenges. 
1.This is about choosing what will be the actual directory,and checking the user running RC process has the proper rights to write there. We may need to modify the Dockerfile image to address this challenge. Something we may investigate during staging test. In any case, this challenge doesnt really involve the scope of this PR, so i guess we can proceed until we face staging test and potentially fix teh Docker image in the proper repo.
2.Plese see new commits updating autoConfig scripts to allow configuration of path.

**Pull request merged**